### PR TITLE
Enable scale subresource on eligible resources

### LINKED ISSUE #4547: Enable scale subresource
When applicable a custom resource should [expose the scale subresource](https://book.kubebuilder.io/reference/generating-crd.html#scale) to let the HPA controller updates the number of replicas.

### PR DESCRIPTION
**Pull request by barkbay**
This PR enables scale subresource on the following resources:
* ApmServer
* Kibana
* EnterpriseSearch
* ElasticMapsServer

If you want to give it a try:

* Update the CRDs with the ones provided by this PR
* Deploy the following resources:

```yaml
apiVersion: elasticsearch.k8s.elastic.co/v1
kind: Elasticsearch
metadata:
  name: elasticsearch-sample
spec:
  version: 7.13.1
  nodeSets:
    - name: default
      count: 1
      config:
        node.store.allow_mmap: false
---
apiVersion: kibana.k8s.elastic.co/v1
kind: Kibana
metadata:
  name: kibana-sample
spec:
  version: 7.13.1
  count: 1
  elasticsearchRef:
    name: "elasticsearch-sample"
  podTemplate:
    spec:
      containers:
        - name: kibana
          resources:
            requests:
              memory: 1Gi
              cpu: 1
            limits:
              memory: 1Gi
---
apiVersion: autoscaling/v2beta2
kind: HorizontalPodAutoscaler
metadata:
  name: kb
spec:
  scaleTargetRef:
    apiVersion: kibana.k8s.elastic.co/v1
    kind: Kibana
    name: kibana-sample
  minReplicas: 1
  maxReplicas: 4
  metrics:
    - type: Resource
      resource:
        name: cpu
        target:
          type: Utilization
          averageUtilization: 50
```

* Stress the CPU by running the following command in the Kibana Pod: `sha1sum /dev/zero`

The deployment should scale up automatically after a few seconds.

Fix #4547

### CODE REVIEW

**Comment by barkbay** [comment]
jenkins test this please

**Comment by barkbay** [comment]
jenkins test this please

File: test/e2e/test/checks.go [hunk 1]
```diff
@@ -74,3 +75,20 @@ func CheckSecretsContent(k *K8sClient, namespace string, expected func() []Expec
...
+	}
+	actualLabels := labelSelector.MatchLabels
+	if len(expectedLabels) != len(actualLabels) {
+		return fmt.Errorf("expected %d labels in selector, got %d instead", len(expectedLabels), len(actualLabels))
+	}
+	for label, expectedValue := range expectedLabels {
+		if actualLabels[label] != expectedValue {
+			return fmt.Errorf("expected selector value for label %s is %s, got %s instead", label, expectedValue, actualLabels[label])
+		}
+	}
```
Comment (pebrc) [inline test/e2e/test/checks.go:92]: Could this just be `deep.Equal(expectedLabels, actualLabels)`?

**Review by pebrc — APPROVED** [review]
LGTM! I did a simple test with Enterprise Search 👍

**Comment by pebrc** [comment]
One more thought: should we mention this new capability in the documentation?

**Comment by barkbay** [comment]
> One more thought: should we mention this new capability in the documentation?

👍 , just opened https://github.com/elastic/cloud-on-k8s/issues/4608, I'll take care of it for `1.7.0`

**Pull request merged**